examples/guidelines-snap-tutorials.md

```diff
@@ -145,190 +255,384 @@ Survey
 
-Of course, this doesn’t concern key tutorial actions happening on a given website, like launchpad, snapweb, juju charm UI…
+Of course, this doesn’t concern key tutorial actions happening on a given website, like Launchpad, GitHub, login.ubuntu.com, etc.
 
-#### Provide regular code “checkpoints”
+### Provide regular code “checkpoints”
 
 This is mostly for developer oriented tutorials, but if this rule can be applied to an admin/advanced users oriented one, please do so!
 
-The advice is to provide regular checkpoints where people can sync back their source code with the tutorial, in particular at the start of a given step. Remember that the attendee can resume a tutorial, and may not have the source code from previous steps anymore. Consequently, even if the previous step had no code change, please do provide back the same link at the start of the following step.
+The advice is to provide regular checkpoints where people can sync back their source code with the tutorial, in particular at the start of a given step. Attendee could have stopped then resumed going through a tutorial, and may not have the source code from previous steps anymore. Consequently, in the case of a code-writing tutorial, please try to provide a link to the current state of the code at the start of a step.
 
-You can refer to it as “If you want to ensure you have all the needed bits we saw together in the previous steps…“ and point to an external repository on github, launchpad or others… This element should be in a note like:
+You can refer to it as “If you want to ensure you have all the needed bits we saw together in the previous steps…“ and point to an external repository on GitHub, launchpad or others… This element should be in an admonition:
 
 Positive
 : **Lost or starting from here?**
 Check or download [here](http://link/torepo/directory) to see what your current directory should look like.
 
-\<here\> being a link to a subdirectory from a git repository. Try to gather similar tutorials in the same repository.
+"here" being a link to the code in question.
+
+### Do not separate exercises and answers
+
+In general, try to avoid leaving the user hanging for the right answer. Tutorials aren’t a class/lesson or a test to pass with some exercises.
 
-The second to last step should give a quick way to experience the final tutorial result, being the entirety of the code or a demo.
+In particular, to avoid a teacher/student relationship, do not separate questions and answers (apart from cliffhangers as previously stated, but the questions you are asking your audience are more rhetorical in that case!).
 
-#### Do not separate exercises and answers
+In general, do not put actionable attendee’s expectations in different steps, questions and answers should be on the same page. This way, the answer is always in the context of the given question.
```

Comment:
I'd considering cutting this, or removing the first part: *In general, do not put actionable attendee’s expectations in different steps*